components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -751,12 +770,237 @@ impl Interpolate for LengthOrNone {
             }
         } else {
             // TODO(gw): Implement matrix decomposition and interpolation
+            // TODO(canaltinova): Do I need to interpolate each item of the list?
             result.extend_from_slice(from_list);
         }
 
         TransformList(Some(result))
     }
 
+    fn rotate_to_matrix(x: f32, y: f32, z: f32, a: SpecifiedAngle) -> ComputedMatrix {
+        let rad = a.radians();
+        let sc = (rad / 2.0).sin() * (rad / 2.0).cos();
+        let sq = 1.0 / 2.0 * (1.0 - (rad).cos());
+
+        ComputedMatrix {
+            m11: 1.0 - 2.0 * (y * y + z * z) * sq, m12: 2.0 * (x * y * sq - z * sc),
+            m13: 2.0 * (x * z * sq + y * sc), m14: 0.0, m21: 2.0 * (x * y * sq + z * sc),
+            m22: 1.0 - 2.0 * (x * x + z * z) * sq, m23: 2.0 * (y * z * sq - x * sc), m24: 0.0,
+            m31: 2.0 * (x * z * sq - y * sc), m32: 2.0 * (y * z * sq + x * sc),
+            m33: 1.0 - 2.0 * (x * x + y * y) * sq, m34: 0.0, m41: 0.0, m42: 0.0, m43: 0.0, m44: 1.0
+        }
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct DecomposedMatrix {
```

Comment:
I tried it but rust is not allowing structs to start with integer value(It's same with MatrixDecomposed2D. I tried 2DMatrixDecomposed but couldn't make it), so I'll change it to `InnerMatrix2D`,  if it's ok.